astropy/io/fits/tests/test_fitscheck.py

```diff
@@ -0,0 +1,115 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import logging
+import os
+import tempfile
+from contextlib import contextmanager
+
+from . import FitsTestCase
+from ..scripts import fitscheck
+from ... import fits
+from ....tests.helper import pytest
+
+
+@contextmanager
+def capture_stderr():
+    log = logging.getLogger('fitscheck')
+    logfile = tempfile.mkstemp(suffix='.log', prefix='fitscheck')[1]
+    handler = logging.FileHandler(logfile)
+    handler.setFormatter(logging.Formatter('%(message)s'))
+    log.addHandler(handler)
+    yield logfile
+    log.removeHandler(handler)
+    os.remove(logfile)
+
+
+class TestFitscheck(FitsTestCase):
+    # def setup(self):
```

Comment:
Just as a reminder to remove this commented code again ... later.

Reply:
Oops yeah, once I find a proper way to capture stderr I will cleanup :wink:

Reply:
Can't you just assign to [`sys.stderr`](https://docs.python.org/3/library/sys.html#sys.stderr)?

Reply:
That's what `capsys` do, but it was failing : there are issues specific to windows and capturing stderr (and it still fails with the last try), and probably a real failure on python 2 (on CircleCI).

Reply:
Ah, do you know which commit that was?

Reply:
The commit using `capsys` is 316fef26d7c6bae75cf80f86ca8592b2de1b59f9 , and appveyor is linked below.